docs/semgrep-ci/configuration-reference.md

```diff
@@ -22,124 +22,243 @@ Object.entries(frontMatter).filter(
 
 # Continuous Integration (CI) configuration reference
 
-Configure Semgrep CI by passing various environment variables in your Continuous Integration (CI) jobs.
+Use this reference to configure Semgrep's behavior in CI environments. This reference includes both command-line options and environment variables. You can also define these variables within a CI configuration file. Refer to your CI provider's documentation for the correct syntax. Examples are written in a Bash environment except when otherwise stated.
 
-:::info
-While environment variables are the preferred way to configure Semgrep CI, pass any of these options as command-line options. Refer to the output of `semgrep ci --help` to find the corresponding flags.
+:::tip
+Unless otherwise stated, you can also set many of these environment variables within your local environment for testing or experimental purposes. Run `semgrep ci` in your command line after setting environment variables to test their effect.
```

Comment:
Adam - we agreed this would be a reference doc, right? It's not meant to tell the user how to set up env vars, it's meant to supplement what the user already knows to do. See https://semgrep.dev/docs/semgrep-ci/running-semgrep-ci-with-semgrep-app/ or https://semgrep.dev/docs/semgrep-ci/running-semgrep-ci-without-semgrep-app/. The user is likely to go through those two first before looking in a reference doc.

Maybe what you'd like is a link instead to either of those guides?

Reply:
Links might help here! @s-santillan I mean - this is not only about reference or not - what do we mean here by the local environment? Going through both guides linked above, I cannot figure out how to include a local CI with environment variables file (without any SCM if I understand that correctly) and run CI locally. 

In short: Do we want to say here that users can try out any configuration of environment variables (for any SCM provider) also locally, by calling respective configuration files (can it be `.semgrep` in the root of a repository https://semgrep.dev/docs/semgrep-ci/running-semgrep-ci-without-semgrep-app/#adding-rules-to-scan-with-semgrep-ci)? Can we mention this in the guides linked above? What would be the commands to do so? Is it `semgrep ci semgrep.yml` or `semgrep ci .gitlab-ci.yml` for respective SCM file? This can be two lines of added text...

Reply:
So - part of your concern should be addressed. But part of your confusion stems from a misunderstanding of CI and env vars. You don't need any CI provider (and therefore you don't need any yaml file, none of those .semgrep-ci.yml) for env vars. Env vars are just env vars. They can be set in a CI provider's environment (remote) and yours (locally). The idea is that the user can set these locally in their development machine to test so they have an idea of how semgrep works when they DO hook it up to a pipeline and run in a remote environment provided by a CI provider. Now - am I worried that **developers** will not understand what I am writing about? Devs should generally know the difference between setting an env var locally and remote (such as CI).

There is no CircleCI, GitHubActions, etc. when the user sets up env vars locally. That's what the word "locally" means in this context.

If these env vars can be set locally, why aren't they under Semgrep CLI? Because the env vars CAN be set locally to for testing purposes to simulate how semgrep will behave in CI, but are MAINLY used in CI.

I think the actionable change here for me is to spell out "local development environment" instead of "local environment" and specify that the env vars are set in the command line.

Reply:
@s-santillan thank you for the explanation! I think that DevOps and devs experienced in working with this can understand this, but not most of them. I am still not sure if I should create a file for these env vars (different than the ones mentioned above for SCMs), or if I can make a oneliner or both previous options together.

Maybe the actionable item can be to document a small procedural article: Testing Semgrep in CI locally? It can be nice for some users...